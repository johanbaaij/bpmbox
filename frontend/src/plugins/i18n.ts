import Vue from "vue";
import VueI18n, { LocaleMessages } from "vue-i18n";
import localesEn from "@/locales/en.json";

Vue.use(VueI18n);

const en: LocaleMessages = { en: localesEn };

export default new VueI18n({
  locale: process.env.VUE_APP_I18N_LOCALE || "en",
  fallbackLocale: process.env.VUE_APP_I18N_FALLBACK_LOCALE || "en",
  messages: en
});
