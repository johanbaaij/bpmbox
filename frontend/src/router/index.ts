import Vue from "vue";
import VueRouter from "vue-router";
import Home from "@/views/Home.vue";
import Register from "@/views/Register.vue";
import Login from "@/views/Login.vue";
import FourOhFour from "@/views/404.vue";
import Dashboard from "@/views/Dashboard.vue";
import EventBus from "@/utils/event-bus";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "home",
    component: Home
  },
  {
    path: "/404",
    name: "FourOhFour",
    component: FourOhFour
  },
  {
    path: "/register",
    name: "register",
    component: Register,
    meta: { auth: false }
  },
  {
    path: "/login",
    name: "login",
    component: Login,
    meta: { auth: false }
  },
  {
    path: "/dashboard",
    name: "dashboard",
    component: Dashboard,
    meta: { auth: true }
  }
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes
});

// add notification to "show on next page queue"
// before navigating, add notification to currently

router.beforeEach((_to, _from, next) => {
  next();
  EventBus.$emit("DISPLAY_NEXT_NOTIFICATION");
});

export default router;
