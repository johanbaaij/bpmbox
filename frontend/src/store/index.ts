import Vue from "vue";
import Vuex from "vuex";
import axios from "axios";
import { jsonapiModule } from "jsonapi-vuex";

Vue.use(Vuex);

const api = axios.create({
  baseURL: process.env["VUE_APP_API_URL"],
  headers: {
    "Content-Type": "application/vnd.api+json"
  }
});

export default new Vuex.Store({
  state: {
    receiving: false
  },
  mutations: {
    receiving(state, payload) {
      state.receiving = payload;
    }
  },
  actions: {
    receiving(context, value) {
      context.commit("receiving", value);
    }
  },
  modules: {
    jv: jsonapiModule(api, {
      preserveJson: true,
      clearOnUpdate: true
    })
  }
});
