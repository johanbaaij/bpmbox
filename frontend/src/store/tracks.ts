import { Module } from "vuex";
import TrackQueryParams from "@/TrackQueryParams";

const tracksModule: Module<any, any> = {
  namespaced: true,
  state: {
    serverItemsLength: 0,
    apiLoading: false,
    hideNullBpms: false,
    options: {
      page: 1,
      itemsPerPage: 30,
      sortBy: ["release_title"],
      sortDesc: [false],
      groupBy: [],
      groupDesc: [],
      mustSort: true,
      multiSort: false
    }
  },
  actions: {
    options({ commit, dispatch }, payload) {
      commit("options", payload);
      dispatch("apiData");
    },
    hideNullBpms({ commit, dispatch }, payload) {
      commit("hideNullBpms", payload);
      dispatch("apiData");
    },
    async apiData({ state, rootState, dispatch, commit }) {
      commit("apiLoading", true);
      const filters = {
        hideNullBpms: state.hideNullBpms
      };
      const urlParams = new TrackQueryParams(
        state.options,
        filters
      ).urlParams();
      const jsonapiData = {
        _jv: { type: "track" }
      };

      const response = await dispatch(
        "jv/get",
        [
          jsonapiData,
          {
            url: `/collections/${rootState.user.username}/tracks`,
            params: {
              include: "release",
              ...urlParams
            }
          }
        ],
        { root: true }
      );
      const { _jv } = response;
      commit("serverItemsLength", _jv.json.meta.total);
      commit("apiLoading", false);
    }
  },
  mutations: {
    options(state, payload) {
      state.options = payload;
    },
    serverItemsLength(state, payload) {
      state.serverItemsLength = payload;
    },
    apiLoading(state, payload) {
      state.apiLoading = payload;
    },
    hideNullBpms(state, payload) {
      state.hideNullBpms = payload;
    }
  }
};

export default tracksModule;
