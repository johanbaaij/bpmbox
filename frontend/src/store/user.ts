import { Module } from "vuex";

const userModule: Module<any, any> = {
  namespaced: true,
  state: {
    username: null
  },
  actions: {
    username({ commit }, payload) {
      commit("username", payload);
    }
  },
  mutations: {
    username(state, payload) {
      state.username = payload;
    },
    initialise(state) {
      if (localStorage.getItem("userModule")) {
        const userStorage = JSON.parse(
          localStorage.getItem("userModule") as string
        );
        state = Object.assign(state, userStorage);
      }
    }
  }
};

export default userModule;
