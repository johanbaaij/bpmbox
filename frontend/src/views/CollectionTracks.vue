<template>
  <v-skeleton-loader
    :loading="initialLoading"
    transition="fade-transition"
    type="table"
  >
    <v-data-table
      fixed-header
      :loading="receiving || apiLoading"
      :items="tracks"
      :headers="headers"
      :options.sync="options"
      :server-items-length="totalTracks"
      :items-per-page="30"
      dense
      :footer-props="{
        disableItemsPerPage: true,
        itemsPerPageOptions: [30]
      }"
    >
      <template v-slot:item.release="{ item }">
        {{ item.release.title }}
      </template>
    </v-data-table>
  </v-skeleton-loader>
</template>

<script lang="ts">
import { mapState } from "vuex";
import { Component, Vue, Prop, Watch } from "vue-property-decorator";
import MetaInfo from "vue-meta";
import TrackQueryParams from "@/TrackQueryParams";

@Component({
  name: "collection-tracks",
  metaInfo(): MetaInfo {
    return {
      title: this.$i18n.t("Collection.title") as string
    };
  },
  computed: {
    ...mapState(["receiving"])
  }
})
export default class CollectionTracks extends Vue {
  @Prop() readonly username!: string;

  initialLoading = true;
  apiLoading = false;
  receiving!: boolean;
  totalTracks = 0;
  tracks = [];

  options = {
    page: 1,
    itemsPerPage: 30,
    sortBy: ["release"],
    sortDesc: [false],
    groupBy: [],
    groupDesc: [],
    mustSort: true,
    multiSort: false
  };

  headers = [
    {
      text: "Position",
      value: "position",
      sortable: false
    },
    {
      text: "Release",
      value: "release",
      sortable: true,
      sort: (a: any, b: any) => {
        if (a.title == b.title) return 0;
        if (a.title > b.title) return 1;
        return -1;
      }
    },
    {
      text: "Artist",
      value: "artist",
      sortable: true
    },
    {
      text: "Title",
      value: "title",
      sortable: true
    },
    {
      text: "BPM",
      value: "bpm",
      sortable: true
    },
    {
      text: "Key",
      value: "key",
      sortable: true
    }
  ];

  @Watch("options", { deep: true })
  optionsChanged(newOptions: any, oldOptions: any) {
    this.getApiData(newOptions);
  }

  async getApiData(options: any) {
    this.apiLoading = true;
    const urlParams = new TrackQueryParams(options).urlParams();
    const jsonapiData = { _jv: { type: "track" } };

    const response = await this.$store.dispatch("jv/get", [
      jsonapiData,
      {
        url: `/collections/${this.username}/tracks`,
        params: { include: "release", ...urlParams }
      }
    ]);

    const { _jv, ...data } = response;

    this.tracks = Object.values(data);

    this.totalTracks = response._jv.json.meta.total;
    // const data = response;
    // this.trracks = data.map(a => data[a.id]);

    this.apiLoading = false;
    this.initialLoading = false;
  }

  async mounted() {
    this.getApiData(this.options);
  }

  // get tracks() {
  //   return Object.values(
  //     this.$store.getters["jv/get"](
  //       "track",
  //       `$[?(@.collection=="${this.username}")]`
  //     )
  //   );
  // }
}
</script>

<style lang="sass"></style>
