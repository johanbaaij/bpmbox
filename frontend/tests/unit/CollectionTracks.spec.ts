import "./setup";

// Dependencies
import Vuetify from "vuetify";
import i18n from "@/plugins/i18n";
import router from "@/router";
import store from "@/store";

// Components
import CollectionTracks from "../../src/views/CollectionTracks.vue";

// Utilities
import { mount, createLocalVue } from "@vue/test-utils";

const localVue = createLocalVue();

describe("CollectionTracks.vue", () => {
  let vuetify: any;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it("should match snapshot", () => {
    const wrapper = mount(CollectionTracks, {
      localVue,
      router,
      vuetify,
      i18n,
      store
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
