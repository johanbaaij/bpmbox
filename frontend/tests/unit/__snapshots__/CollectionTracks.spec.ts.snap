// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionTracks.vue should match snapshot 1`] = `
<div aria-busy="true" aria-live="polite" role="alert" class="v-skeleton-loader v-skeleton-loader--is-loading theme--light">
  <div class="v-skeleton-loader__table v-skeleton-loader__bone">
    <div class="v-skeleton-loader__table-heading v-skeleton-loader__bone">
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
    </div>
    <div class="v-skeleton-loader__table-thead v-skeleton-loader__bone">
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__heading v-skeleton-loader__bone"></div>
    </div>
    <div class="v-skeleton-loader__table-tbody v-skeleton-loader__bone">
      <div class="v-skeleton-loader__table-row-divider v-skeleton-loader__bone">
        <div class="v-skeleton-loader__table-row v-skeleton-loader__bone">
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
        </div>
        <div class="v-skeleton-loader__divider v-skeleton-loader__bone"></div>
      </div>
      <div class="v-skeleton-loader__table-row-divider v-skeleton-loader__bone">
        <div class="v-skeleton-loader__table-row v-skeleton-loader__bone">
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
        </div>
        <div class="v-skeleton-loader__divider v-skeleton-loader__bone"></div>
      </div>
      <div class="v-skeleton-loader__table-row-divider v-skeleton-loader__bone">
        <div class="v-skeleton-loader__table-row v-skeleton-loader__bone">
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
        </div>
        <div class="v-skeleton-loader__divider v-skeleton-loader__bone"></div>
      </div>
      <div class="v-skeleton-loader__table-row-divider v-skeleton-loader__bone">
        <div class="v-skeleton-loader__table-row v-skeleton-loader__bone">
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
        </div>
        <div class="v-skeleton-loader__divider v-skeleton-loader__bone"></div>
      </div>
      <div class="v-skeleton-loader__table-row-divider v-skeleton-loader__bone">
        <div class="v-skeleton-loader__table-row v-skeleton-loader__bone">
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
        </div>
        <div class="v-skeleton-loader__divider v-skeleton-loader__bone"></div>
      </div>
      <div class="v-skeleton-loader__table-row-divider v-skeleton-loader__bone">
        <div class="v-skeleton-loader__table-row v-skeleton-loader__bone">
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
          <div class="v-skeleton-loader__table-cell v-skeleton-loader__bone">
            <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
          </div>
        </div>
        <div class="v-skeleton-loader__divider v-skeleton-loader__bone"></div>
      </div>
    </div>
    <div class="v-skeleton-loader__table-tfoot v-skeleton-loader__bone">
      <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__text v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
      <div class="v-skeleton-loader__avatar v-skeleton-loader__bone"></div>
    </div>
  </div>
</div>
`;
